toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -186,6 +189,35 @@ func getBuildDependencies(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, g
 	return
 }
 
+func parseCheckSection(logFile string) (checkErr error) {
+	file, logErr := os.Open(logFile)
+	// If we can't open the log file, that's a build error.
+	if logErr != nil {
+		logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logFile, logErr)
+		checkErr = logErr
+		return
+	}
+	defer file.Close()
+	for scanner := bufio.NewScanner(file); scanner.Scan(); {
+		currLine := scanner.Text()
+		// Anything besides 0 is a failed test
+		if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
+			failedLogFile := fmt.Sprintf("%s-FAILED_TEST-%d.log", logFile[:strings.Index(logFile, ".log")], time.Now().UnixMilli())
```

Comment:
Nit-pick: while I like one-liners, I think we're overdoing it here a bit and lose clarity in the end. Here's my suggestion to break it down a bit:
```suggestion
			failedLogFile := strings.TrimSuffix(logFile, ".log")
			failedLogFile = fmt.Sprintf("%s-FAILED_TEST-%d.log", failedLogFile, time.Now().UnixMilli())
```

I also replaced using string indeces in favour of `strings.TrimSuffix` to make it clearer what our goal here was.